### ISSUE TITLE
Segfault on astro deploy when no deployments in workspace

### ISSUE THREAD

**Issue by collinmcnulty** [astronomer/astro-cli#1301]
**Describe the bug**

I cannot astro deploy after upgrading to 1.17

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x268 pc=0x1ca9dfe]

goroutine 1 [running]:
github.com/astronomer/astro-cli/cloud/workspace.GetWorkspaces({0x0, 0x0})
        /go/src/github.com/astronomer/astro-cli/cloud/workspace/workspace.go:382 +0x1de
github.com/astronomer/astro-cli/cloud/deployment.Create({0x0, 0x0}, {0xc000472ac0, 0x1a}, {0x0, 0x0}, {0x0, _}, {0xc00045add8, 0x5}, ...)
        /go/src/github.com/astronomer/astro-cli/cloud/deployment/deployment.go:211 +0x291
github.com/astronomer/astro-cli/cloud/deployment.deploymentSelectionProcess({_, _}, {_, _, _}, {_, _}, {_, _})
        /go/src/github.com/astronomer/astro-cli/cloud/deployment/deployment.go:938 +0x3b4
github.com/astronomer/astro-cli/cloud/deployment.GetDeployment({_, _}, {_, _}, {_, _}, _, {_, _}, {0x0, ...})
        /go/src/github.com/astronomer/astro-cli/cloud/deployment/deployment.go:902 +0x371
github.com/astronomer/astro-cli/cloud/deploy.getDeploymentInfo({0x0?, 0x106504e?}, {0xc000472ac0, 0x1a}, {0x0, 0x0}, 0x0, {0xc00019c440, 0xd}, {0x2329a50, ...})
        /go/src/github.com/astronomer/astro-cli/cloud/deploy/deploy.go:369 +0x27f
github.com/astronomer/astro-cli/cloud/deploy.Deploy({{0xc0000440c4, 0x2a}, {0x0, 0x0}, {0xc000472ac0, 0x1a}, {0x20c91b9, 0x5}, {0x20c795a, 0x4}, ...}, ...)
        /go/src/github.com/astronomer/astro-cli/cloud/deploy/deploy.go:220 +0x21d
github.com/astronomer/astro-cli/cmd/cloud.deploy(0xc0000f8900, {0x2c80b88?, 0x0?, 0x0?})
        /go/src/github.com/astronomer/astro-cli/cmd/cloud/deploy.go:142 +0x476
github.com/spf13/cobra.(*Command).execute(0xc0000f8900, {0x2c80b88, 0x0, 0x0})
        /go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000040c00)
        /go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
main.main()
        /go/src/github.com/astronomer/astro-cli/main.go:16 +0x31
```

**What CLI Version did you experience this bug?**

Astro CLI Version: 1.17.0

**This CLI bug is related to which Astronomer Platform?**

- [X] Astro
- [ ] Software
- [ ] None/Unknown

**What Operating System is the above CLI installed on?**

Intel Macbook

**🪜 Steps To Reproduce**

astro deploy

**Comment by collinmcnulty** [comment]
I found the problem. The workspace is real but doesn’t have any deployments in it because I made the deployment in the wrong workspace by accident. Still, it shouldn't segfault.

**Comment by kushalmalani** [comment]
This should be fixed now and will be part of 1.18 release next week
